azure/datalake/store/multithread.py

```diff
@@ -468,14 +468,19 @@ def put_chunk(adlfs, src, dst, offset, size, buffersize, blocksize, delimiter=No
         with adlfs.open(dst, 'wb', blocksize=buffersize, delimiter=delimiter) as fout:
             end = offset + size
             miniblock = min(size, blocksize)
-            with open(src, 'rb') as fin:
-                for o in range(offset, end, miniblock):
-                    if shutdown_event and shutdown_event.is_set():
-                        return nbytes, None
-                    data = read_block(fin, o, miniblock, delimiter)
-                    nwritten = fout.write(data)
-                    if nwritten:
-                        nbytes += nwritten
+            # For empty files there is no need to take the IO hit.
+            if size != 0:
+                with open(src, 'rb') as fin:
+                    for o in range(offset, end, miniblock):
+                        if shutdown_event and shutdown_event.is_set():
+                            return nbytes, None
+                        data = read_block(fin, o, miniblock, delimiter)
+                        nwritten = fout.write(data)
+                        if nwritten:
+                            nbytes += nwritten
+            else:
+                # need to explicitly flush the file in order to create it on the server when the file is empty.
+                fout.flush();
```

Comment:
We do have the close functionality built in for this which does do the flush. Here I want to be explicit though for readability. Additionally, if anything happened to the file that would cause the "close" operation to behave unexpectedly, I want to ensure that we tried to send the data to the server prior to close.